omniduct/databases/base.py

```diff
@@ -202,19 +183,58 @@ def execute_from_file(self, file, **kwargs):
             Extra parameters to pass on to `execute`.
         """
         with open(file, 'r') as f:
-            query = f.read()
-        return self.query(query, **kwargs)
+            return self.execute(f.read(), **kwargs)
 
     def query_from_file(self, file, **kwargs):
         '''
         This method is shorthand for:
         QueryClient.execute_from_file(file, parse=True, **kwargs)
         '''
-        return self.execute_file(file, parse=True, **kwargs)
+        with open(file, 'r') as f:
+            return self.query(f.read(), **kwargs)
 
-    # Uploading data to data store
+    def add_template(self, name, body):
```

Comment:
I'm avoiding writing too much documentation until this all settles down :). But roughly speaking you can do:

```
%%presto.template <template_name>

{{{ <other template name to sub in here }}}

TEMPLATE SQL
```

```%presto <template_name>``` (which uses the global namespace of your current jupyter notebook kernel)